src/js/components/Text/doc.js

```diff
@@ -22,6 +23,47 @@ export const doc = Text => {
     color: colorPropType.description(
       'A color identifier to use for the text color.',
     ),
+    margin: PropTypes.oneOfType([
+      PropTypes.oneOf(['none', ...MARGIN_SIZES]),
+      PropTypes.shape({
+        bottom: PropTypes.oneOfType([
+          PropTypes.oneOf(MARGIN_SIZES),
+          PropTypes.string,
+        ]),
+        end: PropTypes.oneOfType([
+          PropTypes.oneOf(MARGIN_SIZES),
+          PropTypes.string,
+        ]),
+        horizontal: PropTypes.oneOfType([
+          PropTypes.oneOf(MARGIN_SIZES),
+          PropTypes.string,
+        ]),
+        left: PropTypes.oneOfType([
+          PropTypes.oneOf(MARGIN_SIZES),
+          PropTypes.string,
+        ]),
+        right: PropTypes.oneOfType([
+          PropTypes.oneOf(MARGIN_SIZES),
+          PropTypes.string,
+        ]),
+        start: PropTypes.oneOfType([
+          PropTypes.oneOf(MARGIN_SIZES),
+          PropTypes.string,
+        ]),
+        top: PropTypes.oneOfType([
+          PropTypes.oneOf(MARGIN_SIZES),
+          PropTypes.string,
+        ]),
+        vertical: PropTypes.oneOfType([
+          PropTypes.oneOf(MARGIN_SIZES),
+          PropTypes.string,
+        ]),
+      }),
+      PropTypes.string,
+    ]).description(`The amount of margin around the component. For margin to be 
+      applied, Text needs to be wrapped in a containing Box. An object can be 
```

Comment:
Do you think that explicitly specifying Box is a clear way to communicate it? I'm thinking it can be a div, or any other layout component.

Reply:
I think another way to get the margin to take effect is to do something like `as="div"`.

Reply:
You're right, let me clarify the wording here so it is less restricted to "Box"